lib/ZnapZend/Config.pm

```diff
@@ -177,11 +177,16 @@ my $checkBackupSets = sub {
             }
         }
         if ($backupSet->{src_mbuffer}) {
-            if (!($self->zfs->fileExistsAndExec($backupSet->{src_mbuffer}))) {
+            if ($backupSet->{src_mbuffer} ne 'off' and !($self->zfs->fileExistsAndExec($backupSet->{src_mbuffer}))) {
                 warn "*** WARNING: executable '$backupSet->{src_mbuffer}' does not exist on source system, will ignore\n\n";
                 $backupSet->{src_mbuffer} = undef;
             }
         }
```

Comment:
how about assigning  `if (my $src_mbuffer = $backupSet->{src_mbuffer}) {`

Reply:
Maybe my perlish is flaky... but what effect would this give? :)

Especially if you mean replacing this `if` clause (checking for `off` or exec-ability) with that one (checking for `undef`?) and making a new variable no other code knows about? :\

Reply:
you are using `$backupSet->{src_mbuffer}` in side the if then branch ... I often use the pattern:

```perl
if (my $short = $long_potentially_undef_variable){
   do_something_with($short);
   print $short;
   do_something_more($short);
}
```
  
 `$short` is only valid inside the then branch

Reply:
Ah, ok. In this case I think it is more maintainable as is - we check that the variable is initially valid-looking (a setting from ZFS properties), but has values we would not be able to use as a pathname, so `undef` if in run-time to ignore later.

Introducing a virtual variable to shorten a couple of usages would IMHO just muddy the waters here.